homeassistant/components/synology_dsm/config_flow.py

```diff
@@ -130,10 +129,8 @@ async def _show_setup_form(
             data_schema = _discovery_schema_with_defaults(user_input)
             description_placeholders = self.discovered_conf
         elif self.reauth_conf:
-            user_input.update(self.reauth_conf)
-            step_id = "reauth"
-            data_schema = _reauth_schema_with_defaults(user_input)
-            description_placeholders = {EXCEPTION_DETAILS: self.reauth_reason}
+            step_id = "reauth_confirm"
```

Comment:
We shouldn't go directly to another step's form. Each step should only use its own form.

Reply:
sorry, but honestly I did not get the point 😕

Reply:
The method that we're in here will be called from the user step and here were moving to the form of the reauth_confirm step. That's not good. Each step is responsible for its own form. We should only move to another step by calling the step method.

Reply:
What do you think about the [config_flow](https://github.com/home-assistant/core/blob/43b83535669816db50945dd465018ecb9c32b426/homeassistant/components/abode/config_flow.py) of abode integration?
Is this a good one to be used as blueprint for the config flow of Synology DSM?
If so, I would re-work it accordingly 🙂

Reply:
It's ok in regards how it moves between steps. It does allow the user to change the username in the reauth form which will lead to a new config entry created or another entry updated if not the same username as the config entry that started the reauth flow. That part doesn't look completely correct or at least will probably make the user confused.

Normally we should only allow the user to re-authenticate the same config entry as started the reauth flow.